app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java

```diff
@@ -102,125 +100,115 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         return rootview;
     }
 
-    private void fillMap() {
-        mFragmentPositionMap = new HashMap<>();
-        mFragmentPositionMap.put(R.id.images_to_pdf, 1);
-        mFragmentPositionMap.put(R.id.qr_barcode_to_pdf, 1);
-        mFragmentPositionMap.put(R.id.view_files, 2);
-        mFragmentPositionMap.put(R.id.rotate_pages, 3);
-        mFragmentPositionMap.put(R.id.add_watermark, 3);
-        mFragmentPositionMap.put(R.id.merge_pdf, 4);
-        mFragmentPositionMap.put(R.id.split_pdf, 4);
-        mFragmentPositionMap.put(R.id.text_to_pdf, 1);
-        mFragmentPositionMap.put(R.id.compress_pdf, 4);
-        mFragmentPositionMap.put(R.id.remove_pages, 5);
-        mFragmentPositionMap.put(R.id.rearrange_pages, 5);
-        mFragmentPositionMap.put(R.id.extract_images, 5);
-        mFragmentPositionMap.put(R.id.view_history, 2);
-        mFragmentPositionMap.put(R.id.pdf_to_images, 5);
-        mFragmentPositionMap.put(R.id.add_password, 3);
-        mFragmentPositionMap.put(R.id.remove_password, 3);
-        mFragmentPositionMap.put(R.id.add_images, 3);
-        mFragmentPositionMap.put(R.id.remove_duplicates_pages_pdf, 4);
-        mFragmentPositionMap.put(R.id.invert_pdf, 4);
-    }
-
     @Override
     public void onAttach(Context context) {
         super.onAttach(context);
         mActivity = (Activity) context;
     }
 
-    private void setNavigationViewSelection(int index) {
-        if (mActivity instanceof MainActivity)
-            ((MainActivity) mActivity).setNavigationViewSelection(index);
-    }
-
     @Override
     public void onClick(View v) {
 
         Fragment fragment = null;
         FragmentManager fragmentManager = getFragmentManager();
         Bundle bundle = new Bundle();
-        highLightNavigationDrawerItem(v);
 
         switch (v.getId()) {
             case R.id.images_to_pdf:
                 fragment = new ImageToPdfFragment();
+                highLightNavigationDrawerItem(R.id.nav_camera);
```

Comment:
i didnt want this much code repition, thats why I created a hashmap.
Is it possible to do your changes using a hashmap only?

Reply:
Yes, any reason to not use Hashmap? Also refactor `highLightNavigationDrawerItem()` to `highlghtNavigationDrawerItem()`.

Reply:
as now we have divided the xml in groups so 
`mNavigationView.getMenu().getItem(index).setChecked(true);` was not working.
and with ` mNavigationView.setCheckedItem(id);` I need the real id of the nav drawer that is to be highlight not it's index.
so basically now we want home fragment to send the id and not index to main activity so that highlighting is proper

Reply:
okay. I'll refractor highLightNavigationDrawerItem() to highlightNavigationDrawerItem().

Reply:
or I can make a hashmap from id of clicked view in home fragment to id of nav drawer item.
this could also work